tests/integration/support/steps.go

```diff
@@ -56,6 +56,13 @@ func (c *ScenarioCtx) ListenerIsAdded(numberOfListeners int, ListenerName string
 func (c *ScenarioCtx) IsCalledWithArguments(apiName string, arguments *gherkin.DocString) error {
 	c.apiOptions.APIName = apiName
 	c.apiOptions.Arguments = arguments.Content
+
+	// Clearing old state of response, eventdispatcher and decision service
+	c.apiResponse = models.APIResponse{}
+	// Only required for unsessioned tests
+	c.clientWrapper.DecisionService.(*optlyplugins.TestCompositeService).ClearListenersCalled()
```

Comment:
HOw come we need to clean up between calls? Can we just generate a new client wrapper or "get" a new client wrapper based on the requestID?

Reply:
This is for scenarios where multiple api's are called one after another. In such cases, listeners_called, dispatched_events and response of former request needs to be cleared before the latter request is executed. We only recreate client-wrapper when step "the datafile is" is called so for the given case, same client-wrapper is used.

Reply:
What I was saying is that we can use the `request-id` from the request params to cached/fetch instances of client wrappers and create the clientWrapper here when we call `IsCalledWithArguments`. That way we don't need to explicitly clean up these components. We already do this in other SDKs for multiple API calls

Reply:
done, can you please review it now.